x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/tree/Node.java

```diff
@@ -218,13 +218,28 @@ public T transformUp(Function<? super T, ? extends T> rule) {
             transformedChildren.add(next);
         }
 
-        return (childrenChanged ? replaceChildren(transformedChildren) : (T) this);
+        return (childrenChanged ? replaceEachChild(transformedChildren) : (T) this);
     }
 
     /**
-     * Replace the children of this node.
+     * Replaces each child of the current node and validates that the number of children before 
+     * and after this operation will stay the same (default invariant property of node transformations).
```

Comment:
The method signature is confusing - replaceEachChild with a list? Does that mean that other methods do not replace _each_ child but rather skip some?
default invariant is an oxymoron - considering the name of the children can change, clearly it is not an invariant.  This is more of a templating  as oppose to enforcing a constraint.

This method is essentially a wrapper around replaceChildren which validates the size of the children to be the same.
How about `replaceSameNumberOfChildren`  or `replaceChildrenSameSize` ?

Reply:
Removed oxymoron, renamed to `replaceChildrenSameSize`.

Reply:
In the spirit of initial observation: `size` suggests a property of each child (what's the size of a child?), rather than of the collective. `...Count` might have been slightly more evocative, but this is a minor observation anyways.